tests/parse.rs

```diff
@@ -0,0 +1,61 @@
+use fancy_regex::Regex;
+
+#[test]
+fn word_boundary_brace_parsing_ok() {
+    // Valid patterns should succeed
+    assert!(Regex::new(r"\b{start}").is_ok());
+    assert!(Regex::new(r"\b{end}").is_ok());
+    assert!(Regex::new(r"\b{start-half}").is_ok());
+    assert!(Regex::new(r"\b{end-half}").is_ok());
```

Comment:
if these were in the parser itself's `test` (sub)module as linked above, we could assert that parsing `\b{start}` produces an `Expr::Assertion(Assertion::LeftWordBoundary)` as opposed to just that constructing a regex with this pattern succeeds. I feel like it would be more beneficial, what do others think?